OutOfSchool/OutOfSchool.WebApi/Controllers/ParentController.cs

```diff
@@ -0,0 +1,165 @@
+        {
+            this.parentService = parentService;
+        }
+
+        /// <summary>
+        /// To get all Parents from DB.
+        /// </summary>
+        /// <returns>List of Parents.</returns>
+        [HttpGet]
+        public ActionResult<IEnumerable<ParentDTO>> GetParent()
+        {
+            try
+            {
+                return this.Ok(this.parentService.GetAll());
+            }
+            catch (Exception ex)
+            {
+                return this.BadRequest(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// To recieve parent with define id.
+        /// </summary>
+        /// <param name="id">Key in table.</param>
+        /// <returns>Parent with define id.</returns>
+        [HttpGet("{id}")]
+        public async Task<ActionResult<ParentDTO>> GetParentById(long id)
+        {
+            try
+            {
+                ParentDTO parentDTO = await this.parentService.GetById(id).ConfigureAwait(false);
+                return this.Ok(parentDTO);
+            }
+            catch (ArgumentException ex)
+            {
+                return this.BadRequest(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// To create new Parent and add to the DB.
+        /// </summary>
+        /// <param name="parentDTO">ParentDTO object that we want to add.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        [Authorize(Roles = "parent,admin")]
+        [HttpPost]
+        public async Task<ActionResult<ParentDTO>> CreateParent(ParentDTO parentDTO)
+        {
+            if (!this.ModelState.IsValid)
+            {
+                return this.BadRequest(this.ModelState);
+            }
+
+            try
+            {
+                ParentDTO parent = await this.parentService.Create(parentDTO).ConfigureAwait(false);
+                return this.CreatedAtAction(
+                    nameof(this.GetParent),
+                    new { id = parent.Id },
+                    parent);
+            }
+            catch (Exception ex)
+            {
+                return this.BadRequest(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// To update Parent entity that already exists.
+        /// </summary>
+        /// <param name="parentDTO">ParentDTO object with new properties.</param>
+        /// <returns>Parent's key.</returns>
+        [Authorize(Roles = "parent,admin")]
+        [HttpPut]
+        public async Task<ActionResult> Update(ParentDTO parentDTO)
+        {
+            if (parentDTO == null)
+            {
+                return this.BadRequest("Entity was null.");
+            }
+
+            if (!this.ModelState.IsValid)
+            {
+                return this.BadRequest(this.ModelState);
+            }
+
+            try
+            {
+               await this.parentService.Update(parentDTO).ConfigureAwait(false);
```

Comment:
we can return entity from Update method, then we won't need to perform extra call GetById